carcara/src/checker/compression/mod.rs

```
@@ -0,0 +1,407 @@
+    left_parent : usize,
+    right_parent : usize,
+    new_commands : Vec<ProofCommand>,
+) -> Vec<Rc<Term>>{
+    let mut current_vec = Vec::new();
+    let mut current = AHashSet::new();
+    match &new_commands[left_parent] {
+        ProofCommand::Step(step_l) => {
+            for i in 0..step_l.clause.len(){
+                current.insert(step_l.clause[i].remove_all_negations());
+                current_vec.push(step_l.clause[i].remove_all_negations());
+            }
+        }
+        ProofCommand::Assume {id: _, term: term_l} => {
+            current.insert(term_l.remove_all_negations());
+            current_vec.push(term_l.remove_all_negations());
+        }
+        _ => {}
+    }
+
+    match &new_commands[right_parent] {
+        ProofCommand::Step(step_r) => {
+            let mut pivot = step_r.clause[0].remove_all_negations();
+            pivot.0 = 0;
+            let mut is_pivot_in_current = true;
+            for i in 0..current_vec.len(){
+                if pivot.1 == current_vec[i].1 && current_vec[i].0 > 0{
+                    is_pivot_in_current = false;
+                }
+            }
+            binary_resolution(pool, &mut current, &step_r.clause, pivot, is_pivot_in_current);
+            let mut new_clause = Vec::new();
+            for i in current{
+                new_clause.push(unremove_all_negations(pool, i));
+            }
+            return new_clause;
+        }
+        ProofCommand::Assume {id: _, term: term_r} => {
+            let new_clause = [Rc::clone(term_r)];
+            let mut pivot = term_r.remove_all_negations();
+            pivot.0 = 0;
+            let mut is_pivot_in_current = true;
+            for i in 0..current_vec.len(){
+                if pivot.1 == current_vec[i].1 && current_vec[i].0 > 0{
+                    is_pivot_in_current = false;
+                }
+            }
+            binary_resolution(pool, &mut current, &new_clause[..], pivot, is_pivot_in_current);
```

Same as before, ignoring the errors from `binary_resolution`.